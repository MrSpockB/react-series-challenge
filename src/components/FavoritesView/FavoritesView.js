import React, { Component } from 'react';
import { connect } from 'react-redux';
import {Container} from "../../App.styled";
import SearchBar from "../SearchBar/SearchBar";
import GifList from "../GifList/GifList";

class FavoritesView extends Component {

  render() {
    const {allGifs} = this.props.favoritesGifs;
    const data = Object.keys(allGifs).map(key => allGifs[key]);
    return (
      <Container>
        <SearchBar />
        <GifList gifs={data}/>
      </Container>
    )
  }
}

const mapStateToProps = (state) => {
  const { favoritesGifs } = state;
  return { favoritesGifs };
};


export default connect(mapStateToProps)(FavoritesView)