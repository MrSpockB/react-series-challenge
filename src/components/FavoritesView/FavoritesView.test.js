import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { FavoritesView } from "./FavoritesView";

describe('FavoritesView component', () => {
  it('should render correctly', () => {
    const output = shallow(
      <FavoritesView
        filterGifsClear={jest.fn}
        favoritesGifs={{ allGifs: {}, filtered: {} }}
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  })
});