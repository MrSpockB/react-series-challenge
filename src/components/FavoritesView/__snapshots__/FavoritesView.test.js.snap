// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailView component should render correctly 1`] = `
<styled.div>
  <SearchBar
    textButton="Filter"
  />
  <Connect(GifList)
    gifs={Array []}
  />
</styled.div>
`;

exports[`FavoritesView component should render correctly 1`] = `
<styled.div>
  <SearchBar
    textButton="Filter"
  />
  <Connect(GifList)
    gifs={Array []}
  />
</styled.div>
`;
